src/libPOW/CMakeLists.txt

```diff
@@ -5,3 +5,7 @@ target_include_directories (POW PUBLIC ${PROJECT_SOURCE_DIR}/src ${G3LOG_INCLUDE
 
 find_library(OPENCL_LIBRARIES OpenCL ENV LD_LIBRARY_PATH)
 target_link_libraries (POW PRIVATE ${OPENCL_LIBRARIES} ethash ethash-cl Constants)
+
+if(CUDA_FOUND)
+    target_link_libraries(POW PRIVATE ethash-cuda)
```

Comment:
can you change to dynamic linking? using the SHARED instead of PRIVATE?

Reply:
The libethash-cuda is created as a SHARED library already. 
cuda_add_library(ethash-cuda SHARED ${sources} ${headers})
The PRIVATE in target_link_libraries means the POW depends on ethash, but ethash not depends on POW.